lib/SIL/Verifier/SILOwnershipVerifier.cpp

```diff
@@ -772,16 +772,9 @@ verifySILValueHelper(const SILFunction *f, SILValue value,
     return;
 
   SmallPtrSet<SILBasicBlock *, 32> liveBlocks;
-  if (deadEndBlocks) {
-    SILValueOwnershipChecker(*deadEndBlocks, value, errorBuilder, liveBlocks,
-                             reborrowVerifier)
-        .check();
-  } else {
-    DeadEndBlocks deadEndBlocks(f);
-    SILValueOwnershipChecker(deadEndBlocks, value, errorBuilder, liveBlocks,
-                             reborrowVerifier)
-        .check();
-  }
+  SILValueOwnershipChecker(*deadEndBlocks, value, errorBuilder, liveBlocks,
```

Comment:
I don't understand why this was a miscompile? The shadowing rules mean the inner DeadEndBlocks is passed.

Reply:
the reason I put this in there is for usage in the debugger, can you add an extra method that doesn't take a DEBlock that has that behavior? I am fine if it is marked as deprecated so it can only be used without a warning in the debugger.

Reply:
`verifySILValueHelper` was fine. I only changed it in this PR because there was no need to work around the `nullptr` default anymore. The problem is that `ReborrowVerifier` takes a reference to DeadEndBlocks which is undefined behavior.

Reply:
Can you call that out in the commit message?

Reply:
@atrick (IOW there is no way from this patch I would have understood that. It would be great to also get the file/some easy way to identify the reference [e.x.: the reborrow verifier has a class called FooContext that stores into its state a reference to this so we can never have a nullptr]). That being said, I still need to think about if verifyOwnership is actually helpful in the debugger. So I am ok with this, given more context is put into the commit message.